relay-general/src/store/schema.rs

```diff
@@ -120,15 +90,17 @@ where
     }
 }
 
-fn verify_value_pattern(
+fn verify_value_characters(
     value: &mut String,
     meta: &mut Meta,
     state: &ProcessingState<'_>,
 ) -> ProcessingResult {
-    if let Some(ref regex_string) = state.attrs().match_regex {
-        if !get_regex(regex_string).is_match(value) {
-            meta.add_error(Error::invalid("invalid characters in string"));
-            return Err(ProcessingAction::DeleteValueSoft);
+    if let Some(ref character_set) = state.attrs().characters {
+        for c in value.chars() {
+            if !character_set.is_match(c) {
```

Comment:
You can "simplify" this to:

```rust
if value.chars().any(|c| !character_set.is_match(c)) {
  // add error here
}
```

This would probably be a good utility function on CharacterSet, taking an `&str`.

Reply:
wanted to show invalid character in error message so I left it like that (could've went with `Result<(), char>` as return type but that's too complicated.

Reply:
I see, that makes sense. It would be nice if we could print the entire valid character set (similar to how often password fields tell you which characters are valid, not which character you entered wrong).